### ISSUE TITLE
rebuild step not recognizing native dependency skia-canvas in cross-compile scenario

### ISSUE THREAD

**Issue by tom2strobl** [electron-userland/electron-builder#7563]
* **Electron-Builder Version**: 24.3.0

* **Node Version**: v18.13.0

* **Electron Version**: 18.2.2
* **Electron Type (current, beta, nightly)**: current

* **Target**: mac-x64 & mac-arm64

Hi there,

I got a case where I have a native dependency `skia-canvas`, it absolutely is listed in the same `dependencies` field of the correct package.json where all my other native dependencies are (eg. keytar, which works just fine). For some reason during packaging, during the `rebuilding native dependencies` step, `skia-canvas` is never listed, hence not correctly rebuilt for my target arch.

Since they publish builds for the different archs, I now simply use a `beforeBuild` step where I just do
```
npm rebuild skia-canvas --target_arch=${context.arch} --update-binary
```
which works, but is quite messy as I think the rebuild step from electron-builder should normally catch it, right? (this has to happen during electron-builders workload since we want to cross-compile for respective target-archs)

In terms of config I'm using
```
npmRebuild: true, // to enable beforeBuild
nodeGypRebuild: false, // since I want to leverage prebuild
buildDependenciesFromSource: true, // settings this to false has no effect on skia-canvas being recognized or not
```

They use [napi v6](https://github.com/samizdatco/skia-canvas/blob/35ac526e8b428579f84b7d9557f032cbf5e04883/package.json#L54) and publish [node-pre-gyp prebuilds](https://github.com/samizdatco/skia-canvas/blob/35ac526e8b428579f84b7d9557f032cbf5e04883/package.json#L24).

It's in my regular `node_modules` folder where the other native deps are as well, so it's not a weird location that's causing it.

They also don't have a weird file extension, it's a regular `.node` file that is being compiled.
<img width="173" alt="image" src="https://user-images.githubusercontent.com/557074/234860825-cf7be189-f305-43d0-bd68-8ec914ba01d5.png">

Could it be that because they don't have a `binding.gyp` file in their repo and electron-builder expects it to be present?

It's also very easy that I misunderstood something 🙃

**Comment by mmaietta** [comment]
electron-builder uses `@electron/rebuild` under the hood in v24. You may want to try using the DEBUG env var (I think it's `DEBUG=electron/rebuild`?) to determine how it's parsing the module tree and why it's not building your native module.

**Comment by tom2strobl** [comment]
Ah, forgot to answer here. `DEBUG=electron/rebuild` had no effect and the readme at https://github.com/electron/rebuild has no further information on any debug vars either.

**Comment by mmaietta** [comment]
Quick note. We've reverted to the v23.6.0 approach of native module compilation + prebuilds. It's available in v24.4.2 IIRC
